packages/@aws-cdk/aws-appmesh/lib/route-spec.ts

```diff
@@ -0,0 +1,241 @@
+   * You can also match for path-based routing of requests. For example, if your virtual service name is my-service.local
+   * and you want the route to match requests to my-service.local/metrics, your prefix should be /metrics.
+   */
+  readonly prefixPath: string;
+}
+
+/**
+ * Properties specific for HTTP Based Routes
+ */
+export interface HttpRouteSpecOptions {
+  /**
+   * The criterion for determining a request match for this Route
+   *
+   * @default - matches on '/'
+   */
+  readonly match?: HttpRouteMatch;
+
+  /**
+   * List of targets that traffic is routed to when a request matches the route
+   */
+  readonly weightedTargets: WeightedTarget[];
+
+  /**
+   * The priority for the route. Routes are matched based on the specified value, where 0 is the highest priority.
+   *
+   * @default - none
+   */
+  readonly priority?: number;
+}
+
+/**
+ * Properties specific for a TCP Based Routes
+ */
+export interface TcpRouteSpecOptions {
+  /**
+   * List of targets that traffic is routed to when a request matches the route
+   */
+  readonly weightedTargets: WeightedTarget[];
+}
+
+/**
+ * All Properties for GatewayRoute Specs
+ */
+export interface RouteSpecConfig {
+  /**
+   * The spec for an http route
+   *
+   * @default - no http spec
+   */
+  readonly httpRouteSpec?: CfnRoute.HttpRouteProperty;
+
+  /**
+   * The spec for an http2 route
+   *
+   * @default - no http2 spec
+   */
+  readonly http2RouteSpec?: CfnRoute.HttpRouteProperty;
+
+  /**
+   * The spec for a grpc route
+   *
+   * @default - no grpc spec
+   */
+  readonly grpcRouteSpec?: CfnRoute.GrpcRouteProperty;
+
+  /**
+   * The spec for a tcp route
+   *
+   * @default - no tcp spec
+   */
+  readonly tcpRouteSpec?: CfnRoute.TcpRouteProperty;
+
+  /**
+   * The priority for the route. Routes are matched based on the specified value, where 0 is the highest priority
+   *
+   * @default - none
+   */
+  readonly priority?: number;
+}
+
+/**
+ * Used to generate specs with different protocols for a RouteSpec
+ */
+export abstract class RouteSpec {
+  /**
+   * Creates an HTTP Based RouteSpec
+   */
+  public static http(options: HttpRouteSpecOptions): RouteSpec {
+    return new HttpRouteSpec(options, Protocol.HTTP);
+  }
+
+  /**
+   * Creates an HTTP2 Based RouteSpec
+   *
+   */
+  public static http2(options: HttpRouteSpecOptions): RouteSpec {
+    return new HttpRouteSpec(options, Protocol.HTTP2);
+  }
+
+  /**
+   * Creates a TCP Based RouteSpec
+   */
+  public static tcp(options: TcpRouteSpecOptions): RouteSpec {
+    return new TcpRouteSpec(options);
+  }
+
+  /**
+   * List of targets that traffic is routed to when a request matches the route
+   */
+  abstract readonly weightedTargets: WeightedTarget[];
+
+  /**
+   * The priority for the route. Routes are matched based on the specified value, where 0 is the highest priority
+   */
+  abstract readonly priority?: number;
```

Comment:
Why are we exposing these - I don't see them used in `Route`...? I think we only need `bind()` in this class, no?

We should strive to minimize the public API surface as much as we can.

Reply:
So, this field makes a bit more sense once we implement header based routing for HTTP.

Essentially it is used to disambiguate when a request matches multiple routes. I'll remove for this PR and we can add it back later!